app/controllers/questionnaire_controller.rb

```diff
@@ -75,6 +76,13 @@ def create_informed_consent
     redirect_to questionnaire_path(uuid: @response.uuid)
   end
 
+  def set_language_preference
+    selected_locale = questionnaire_content['v0']
+    current_user.update! locale: selected_locale == 'Nederlands' ? 'nl' : 'en'
+    @response.invitation_set.update! locale_retrieved: true
+    redirect_to NextPageFinder.get_next_page current_user: current_user
+  end
+
```

Comment:
Wait, so it sets the user locale based on something that is encoded in the questionnaire, and then afterwards it doesn't set it back? I'm not sure if it is the best solution. Couldn't we do something in the `questionnaire_generator.rb` itself that when it sees there's a "v0" key (which I would prefer if we just had maybe a "lang" key in the questionnaires, and leave the `v` prefix for actual questions), but if it had such a key that it then would override its given `locale`  param (see line 13 in `app/generators/questionnaire_generator.rb`). Would that work for this intended purpose?